File: src/test/scala/tests/graphics/TestText.scala
package tests

import sfml.graphics.*
import sfml.window.*

object TestText extends SNTest:
    override def snTest(snTestScreen: TestScreen): Unit =
        scala.util.Using.Manager { use =>
            // Setup
            val window = use(RenderWindow(VideoMode(1024, 768), "Test"))

            val font = use(Font())
            font.loadFromFile("src/test/resources/tuffy.ttf")

            val text = use(Text("Hello World", font, 50))

            window.isOpen()

            // Control test
            window.clear()
            window.draw(text)
            window.display()
            snTestScreen.takeScreenshot()

            // sf::Text::setPosition
            text.position = (100, 100)

            window.clear()
            window.draw(text)
            window.display()
            snTestScreen.takeScreenshot()

            // Teardown
            window.closeWindow()
        }

File: src/test/cxx/tests/graphics/TestText.cpp
#include "Test.hpp"

#include <SFML/Graphics.hpp>
#include <iomanip>
#include <iostream>

snTest(TestText)
{
    // Setup
    sf::RenderWindow window(sf::VideoMode(1024, 768), "Test");

    sf::Font font;
    font.loadFromFile("src/test/resources/tuffy.ttf");

    sf::Text text("Hello World", font, 50);

    window.isOpen();

    // Control test
    window.clear();
    window.draw(text);
    window.display();
    snTestScreen.takeScreenshot();

    // sf::Text::setPosition
    text.setPosition(100, 100);

    window.clear();
    window.draw(text);
    window.display();
    snTestScreen.takeScreenshot();

    // sf::Text::getGlobalBounds
    auto globalBounds = text.getGlobalBounds();

    std::cout << std::fixed << std::setprecision(1)
              << "Global bounds: ("
              << globalBounds.left << ", "
              << globalBounds.top << ", "
              << globalBounds.width << ", "
              << globalBounds.height << ")" << std::endl;

    // Teardown
    window.close();
}
